playground/index.ts

```diff
@@ -1,25 +1,108 @@
+async function getVisitorData() {
   const fp = await FingerprintJS.load()
-  const result = await fp.get({ debug: true })
-  return result.visitorId
+  return await fp.get({ debug: true })
 }
 
 async function startPlayground() {
-  const display = document.querySelector('.visitorId')
-  if (!display) {
-    throw new Error("The display element isn't found in the HTML code")
+  const output = document.querySelector('.output')
+  if (!output) {
+    throw new Error("The output element isn't found in the HTML code")
   }
+
   try {
-    const visitorId = await getVisitorId()
-    display.textContent = visitorId
-    display.classList.remove('loading')
+    const startTime = Date.now()
+    const { visitorId, components } = await getVisitorData()
+    output.innerHTML = ''
+    addOutputSection(output, 'Visitor identifier:', visitorId, 'big')
+    addOutputSection(output, 'Time took to get the identifier:', `${Date.now() - startTime}ms`)
+    addOutputSection(output, 'Entropy components:', FingerprintJS.componentsToDebugString(components), 'small')
+    initializeDebugButtons(`${visitorId}:
+\`\`\`
+${FingerprintJS.componentsToDebugString(components)}
+\`\`\``)
   } catch (error) {
-    display.textContent = 'Unexpected error'
+    const errorData = {
+      message: error.message,
+      stack: error.stack.split('\n'),
+      ...error,
+    }
+    output.innerHTML = ''
+    addOutputSection(output, 'Unexpected error:', JSON.stringify(errorData, null, 2))
+    initializeDebugButtons(`Unexpected error:\n
+\`\`\`
+${JSON.stringify(errorData, null, 2)}
+\`\`\``)
     throw error
   }
 }
 
+function addOutputSection(output: Node, header: string, content: string, size?: 'small' | 'big') {
+  const headerElement = document.createElement('div')
+  headerElement.classList.add('heading')
+  headerElement.textContent = header
+  output.appendChild(headerElement)
+
+  const contentElement = document.createElement('pre')
+  if (size) {
+    contentElement.classList.add(size)
+  }
+  contentElement.textContent = content
+  output.appendChild(contentElement)
+}
+
+function initializeDebugButtons(debugText: string) {
+  const copyButton = document.querySelector('#debugCopy')
+  if (copyButton instanceof HTMLButtonElement) {
+    copyButton.disabled = false
+    copyButton.addEventListener('click', async (event) => {
+      event.preventDefault()
+      copy(debugText)
+    })
+  }
+
+  const shareButton = document.querySelector('#debugShare')
+  if (shareButton instanceof HTMLButtonElement) {
+    shareButton.disabled = false
+    shareButton.addEventListener('click', async (event) => {
+      event.preventDefault()
+      share(debugText)
+    })
+  }
+}
+
+function copy(text: string) {
+  const textarea = document.createElement('textarea')
+  textarea.value = text
+  document.body.appendChild(textarea)
+  textarea.focus()
+  textarea.select()
+  try {
+    document.execCommand('copy')
+  } catch {
+    // Do nothing in case of a share abort
```

Comment:
looks like a copy-pasted comment from the 'share' method

Reply:
You are right, fixed